### PR TITLE
Feature: Svelte SDK

### PR DESCRIPTION
**Pull request by samijaber**
## Description

- extract `RenderInlinedStyles` and reuse for both Content and Block styles
- adds Svelte SDK

Everything in `output/svelte` is generated by the Mitosis function, and can be ignored review-wise.

### CODE REVIEW

File: packages/sdks/mitosis.config.js [hunk 1]
```diff
@@ -11,9 +12,10 @@ const getSeededId = () => {
  */
 module.exports = {
   files: 'src/**',
-  targets: ['reactNative', 'vue', 'solid'],
+  targets: ['reactNative', 'vue', 'solid', 'svelte'],
```
Comment (mrkoreye) [inline packages/sdks/mitosis.config.js:15]: 😎

File: packages/sdks/output/react-native/src/components/render-inlined-styles.lite.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function RenderInlinedStyles(props) {
+  function injectedStyleScript() {
+    return `<${tagName()}>${props.styles}</${tagName()}>`;
+  }
+
+  function tagName() {
+    // NOTE: we have to obfusctate the name of the tag due to a limitation in the svelte-preprocessor plugin.
+    // https://github.com/sveltejs/vite-plugin-svelte/issues/315#issuecomment-1109000027
+    return "sty" + "le";
```
Comment (mrkoreye) [inline packages/sdks/output/react-native/src/components/render-inlined-styles.lite.tsx:13]: 🙈 😆

File: packages/sdks/output/svelte/src/blocks/columns/columns.svelte [hunk 3]
```diff
@@ -0,0 +1,103 @@
+<script>
+
```
Comment (mrkoreye) [inline packages/sdks/output/svelte/src/blocks/columns/columns.svelte:2]: I assume all the wacky spacing in here is due to svelte not playing well with prettier?

File: packages/sdks/src/components/render-block/block-styles.lite.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export default function BlockStyles(props: BlockStylesProps) {
+  const state = useState({
+    camelToKebabCase(string: string) {
+      return string
+        .replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, '$1-$2')
+        .toLowerCase();
+    },
+
+    get css(): string {
+      // TODO: media queries
```
Comment (mrkoreye) [inline packages/sdks/src/components/render-block/block-styles.lite.tsx:18]: Is this a limitation for all the new SDKs, as in do they not support the different device size styles in the Builder editor?

**Review by mrkoreye — APPROVED** [review]

File: packages/sdks/output/svelte/src/blocks/columns/columns.svelte (hunk 3 above)
Comment (samijaber) [reply packages/sdks/output/svelte/src/blocks/columns/columns.svelte:2]: Exactly, the svelte prettier plugin was crashing constantly so I disabled it for now. 

It's on my list of lower-priority things to investigate once the SDK is in a working state.

File: packages/sdks/src/components/render-block/block-styles.lite.tsx (hunk 4 above)
Comment (samijaber) [reply packages/sdks/src/components/render-block/block-styles.lite.tsx:18]: TBH I'm unsure. I copy-pasted this code from here:

https://github.com/BuilderIO/builder/blob/c9b5391bd7edcc17e0b25b03fef13f5c05fc9526/packages/sdks/overrides/vue/src/components/block-styles.vue#L12-L31

It was a Vue-specific override, and I extracted it out to be usable by other frameworks as well.

**Pull request merged**